gensim/corpora/textcorpus.py

```diff
@@ -216,13 +216,31 @@ def __init__(self, input=None, dictionary=None, metadata=False, character_filter
 
         Examples
         --------
-        >>> #TODO Example with inheritance
         >>> from gensim.corpora.textcorpus import TextCorpus
+        >>> from gensim import corpora
         >>> from gensim.test.utils import datapath
+        >>> from gensim import utils
         >>>
-        >>> corpus = TextCorpus(datapath('head500.noblanks.cor.bz2'))
-        >>> for bow in corpus:
-        ...     pass
+        >>> class CorpusMiislita(corpora.TextCorpus):
+        >>>     stoplist = set('for a of the and to in on'.split())
+        >>>
+        >>>     def get_texts(self):
+        >>>         for doc in self.getstream():
+        >>>             yield [word for word in utils.to_unicode(doc).lower().split()
+        >>>                 if word not in CorpusMiislita.stoplist]
+        >>>
+        >>>     def __len__(self):
+        >>>         if 'length' not in self.__dict__:
+        >>>             logger.info("caching corpus size (calculating number of documents)")
+        >>>         self.length = sum(1 for _ in self.get_texts())
+        >>>         return self.length
+        >>>
+        >>> corpus = CorpusMiislita(datapath('head500.noblanks.cor.bz2'))
+        >>> corpus.get_texts()
+        <generator object get_texts at 0x7fa932f397d0>
+        >>> corpus.__len__()
```

Comment:
please `len(dorpus)` instead of this one, call "magic" directly is bad pattern (and is justified only for specific cases)